cmake/dependencies.cmake

```diff
@@ -1,18 +1,28 @@
 include(FetchContent)
 
-if (NOT GRPCXX_USE_ASIO)
-	# libuv
-	FetchContent_Declare(libuv
-		URL      https://github.com/libuv/libuv/archive/refs/tags/v1.46.0.tar.gz
-		URL_HASH SHA256=7aa66be3413ae10605e1f5c9ae934504ffe317ef68ea16fdaa83e23905c681bd
-	)
+option(HERMETIC_BUILD "Fetch and build all dependencies instead of relying on system libraries" ON)
```

Comment:
Can we move this to the root `CMakeLists.txt` please? I think it's better to keep all `option` directives together instead to splitting them across multiple files.

Also, please prefix with `GRPCXX_` so it won't clash with other projects.

Reply:
Done. I also fixed a few more smaller problems in include paths and corner cases.